lib/src/main/java/com/auth0/jwt/JWTCreator.java

```diff
@@ -164,6 +176,78 @@ public Builder withJWTId(String jwtId) {
             return this;
         }
 
+        /**
+         * Add a Map of Claim values. Only primitives and collections are supported as
+         * values in the map. Warning: Custom classes are not allowed.
+         *
+         * @param name
+         * @param values
+         * @throws IllegalArgumentException
+         */
+        public Builder withClaim(String name, Map<String, Object> values) throws IllegalArgumentException {
+            assertNonNull(name);
+            if (!validateClaim(values)) {
+                throw new IllegalArgumentException(
+                        "Unsupported claim type found in passed collection. Refer supported types.");
+            }
+            addClaim(name, values);
+            return this;
+        }
+
+        /**
+         * Add a Collection of Claim values. Only primitives and supported array types
+         * are supported as values in the Collection. Warning: Custom classes are not
+         * allowed.
+         *
+         * @param name
+         * @param values
+         * @throws IllegalArgumentException
+         */
+        public Builder withClaim(String name, Collection<Object> values) throws IllegalArgumentException {
+            assertNonNull(name);
+            if (!validateClaim(values)) {
+                throw new IllegalArgumentException(
+                        "Unsupported claim type found in passed collection. Refer supported types.");
+            }
+            addClaim(name, values);
+            return this;
+        }
+
+        private static boolean validateClaim(Map<String, Object> claim) {
+            if (claim == null || claim.isEmpty()) {
+                return false;
+            }
+            Collection<Object> values = claim.values();
+            return validateClaim(values);
+        }
+
+        private static boolean validateClaim(Collection<Object> values) {
+            if (values == null || values.isEmpty()) {
```

Comment:
values.isEmpty() - does it mean I cannot have a claim with an empty object, like this:

```
"warnings": {}
```

I don't think it is right then. This is a totally valid claim.

Reply:
Agree, removed the empty check and keeping the null check as the library also enforces it.